src/porepy/models/solution_strategy.py

```diff
@@ -203,6 +214,12 @@ def prepare_simulation(self) -> None:
         # Order of operations is important here.
         self.set_equation_system_manager()
         self.create_variables()
+        # After fluid and variables are defined, we can define the secondary quantities like
+        # fluid properties (which depend on variables).
+        # NOTE this is critical in the case where properties depend on some fractions.
+        # The callables for those are dynamically created during create_variables, as opposed
+        # to e.g., pressure or temperature.
+        self.assign_thermodynamic_properties_to_phases()
```

Comment:
This step was added to the set-up.
It is critical that the Fluid is created in `set_materials`, before the variables, since the dynamic number of variables depends on the fluid definition.
The step `assign_thermodynamic_properties_to_phases` requires the variables to be already created, since fluid properties are in general secondary terms which depend on some variable (keyword SurrogateFactory)